lib/resources.js

```diff
@@ -377,15 +377,13 @@ export async function _updateAcquisitionRecord({
     'acquisition.ttl': ttl
   };
   const update = {$set};
-  let dbOptions;
+  const dbOptions = {};
   if(upsert) {
-    dbOptions = {...database.writeOptions, upsert: true};
+    dbOptions.upsert = true;
     update.$setOnInsert = {
       'meta.created': now,
       'acquisition.acquirerId': acquirerId
     };
-  } else {
-    dbOptions = database.writeOptions;
```

Comment:
I can't remember if the above works or not, i.e., if it behaves the same whether we bring in all of the options manually and layer `upsert` on top or if we can just pass `upsert` and it will auto-layer it onto the existing options. I know at some point this didn't work, but I'm hopeful it does now. If someone could hack into the code (down deep in the mongodb driver layer) and print out what it does to confirm that it's the same, that would be ideal.

Reply:
@dlongley Here's what the options look like when it gets to mongodb:
https://gist.github.com/JSAssassin/09b25aa872baadcaa677438f7a88ae5c

Looks the same except `forceServerObjectId` is missing from the one where we only pass upsert.

Reply:
@JSAssassin,

Thanks, being able to see the differences is excellent. I was mostly concerned about `writeConcern` being the same -- and other than some `WriteConcern` wrapper being used in some cases, it looks the same, so I presume we're good there. We don't do anything special with `_id` ... so I don't think we need to worry about `forceServerObjectId` either way. This looks good, thank you!